### PR TITLE
Updating tutorials

### PR DESCRIPTION
**Pull request by fpan017**

### CODE REVIEW

**Comment by fpan017** [comment]
first commit

File: docs/source/tutorials/visualising_scaffolds/visualising_scaffolds.rst [hunk 1]
```diff
@@ -1,21 +1,50 @@
...
+.. image:: ../../images/visualization_scaffold/vs4.png
+    :class: with-shadow
+
+Once you are happy with the shape, click :guilabel:`<< Back` on the top left corner. You should see the shape
+of urethra has already changed.
+
+.. image:: ../../images/visualization_scaffold/vs3.png
+    :class: with-shadow
+
+You may found some meshes are quite big. If you see part of the mesh exceeds the coordinates system's limitation and disappears on screen,
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/visualising_scaffolds/visualising_scaffolds.rst]: "If part or all of the mesh is not visible in the field of view of the scaffoldmaker's graphics viewport, you can rescale the mesh by holding ....

File: docs/source/installation.rst [hunk 2]
```diff
@@ -3,7 +3,7 @@ Getting started
 
 .. toctree::
   :maxdepth: 2
-  :caption: Contents:
```
Comment (PrasadBabarendaGamage) [inline docs/source/installation.rst:6]: Best to leave this as contents to be consistent?

File: docs/source/tutorials/creating_meshtype/creating_3d_cube.rst [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+.. role:: pyth(code)
+  :language: python
+
+Let's create a 3D cube as your first mesh type.
+
+Set-up files and modules
+-------------------------
+
+First, go to where you put :file:`scaffoldmaker`, find the :file:`.\\src\\scaffoldmaker\\meshtypes` folder. This is where you put all mesh types.
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_cube.rst]: First, navigate to where you installed scaffoldmaker, ...

File: docs/source/tutorials/creating_meshtype/creating_3d_cube.rst [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        def generateBaseMesh(cls, region, options):
+            ...
+            mesh = fm.findMeshByDimension(3)
+            tricubichermite = eftfactory_tricubichermite(mesh, False) # False = no cross derivative.
+            eft = tricubichermite.createEftBasic()
+            elementtemplate = mesh.createElementtemplate()
+            elementtemplate.setElementShapeType(Element.SHAPE_TYPE_CUBE)
+            result = elementtemplate.defineField(coordinates, -1, eft)
+
+Continue from above, we then initialize the mesh and define our interpolation function
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_cube.rst]: Continuing

File: docs/source/tutorials/creating_meshtype/creating_3d_cube.rst [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+using :pyth:`.findMeshByDimension()` and :pyth:`eftfactory_tricubichermite()` respectively. :pyth:`eft` refers to element field template, a template that defines
+field parameter mapping and interpolation over an element. If you want to use aa interpolation function that is not included in :ref:`utils`,
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_cube.rst]: aa -> a

File: docs/source/tutorials/creating_meshtype/creating_3d_cube.rst [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+you may have to manually set-up eft using basic functions provided by `Zinc <http://opencmiss.org/zinc/latest/classes.html>`_.
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_cube.rst:113]: set-up->set up for British English
eft?

File: docs/source/tutorials/creating_meshtype/creating_3d_cube.rst [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
+Just like nodes, we also have to create a template for elements using :pyth:`.createElementtemplate()`.
+:pyth:`.setElementShapeType()` defines
+the structure of each element. At last, we use :pyth:`defineField` to define the field components
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_cube.rst:117]: At last, -> Finally,

File: docs/source/tutorials/creating_meshtype/creating_3d_cube.rst [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+            elementIdentifier = 1
+            element = mesh.createElement(elementIdentifier, elementtemplate)
+            element.setNodesByIdentifier(eft, [1, 2, 3, 4, 5, 6, 7, 8])
+
+            fm.endChange()
+            return []
+
+Be careful, the order of node identifier matters. For example, between node 1 and 2, we use the derivatives of the global
+coordinates with respect to the local coordinates in the first direction, e.g. :pyth:`dx_ds1`. Whereas :pyth:`dx_ds2`
+generates mesh in the second direction, i.e, along node 1 and node 3. :pyth:`dx_ds3` is the derivatives corresponding to the third
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_cube.rst:144]: generates a mesh

File: docs/source/tutorials/creating_meshtype/creating_3d_tube.rst [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+    from opencmiss.zinc.field import Field
+    from opencmiss.zinc.node import Node
+    from scaffoldmaker.meshtypes.scaffold_base import Scaffold_base
+    from scaffoldmaker.utils.eftfactory_tricubichermite import eftfactory_tricubichermite
+
+Creating user interface options
+-------------------------------
+
+As mentioned, this time we will let the user decide the number of elements in each direction.
+Therefore we should set-up the user interface options as followed:
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_tube.rst:27]: set-up->set up for British English

File: docs/source/tutorials/creating_meshtype/creating_3d_tube.rst [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+                if options[key] < 1:
+                    options[key] = 1
+            if (options['Number of elements around'] < 3) :
+                options['Number of elements around'] = 3
+            if (options['Wall thickness'] < 0.0) :
+                options['Wall thickness'] = 0.0
+            elif (options['Wall thickness'] > 0.5) :
+                options['Wall thickness'] = 0.5
+
+Everything is pretty similar to what we have done in the :ref:`3D cube <cube>` tutorial. Feel free to add your own options or constraints.
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_tube.rst:35]: remove the word "pretty"
Best if we don't add any extra adjectives as they are not typically used in scientific writing

File: docs/source/tutorials/creating_meshtype/creating_3d_tube.rst [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+            return []
+
+If you look back, our node identifiers accumulate in the following order:
+
+* along the tube :pyth:`e1`
+* through the wall :pyth:`e2`
+* around the tube :pyth:`e3`
+
+Thus, we can address the first node identifier in each element, then use the first node identifier to locate others. In the
+last iteration of :pyth:`e3`, nodes should return to the starting position. Thus I subtracted :pyth:`totalNodes`
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_tube.rst]: Thus I -> Thus we

File: docs/source/tutorials/creating_meshtype/creating_3d_tube.rst [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+from each node identifier if it exceeds the total number of nodes, to complete the loop. If you are struggling
+to picture this, check the figure below for better understanding.
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/creating_meshtype/creating_3d_tube.rst]: check the figure below for better understanding ->see the figure below for more information.

File: docs/source/tutorials/visualising_scaffolds/visualising_scaffolds.rst [hunk 13]
```diff
@@ -1,21 +1,50 @@
-*******
-Example
-*******
+Example of visualising scaffolds
+================================
 
-.. toctree::
-   :maxdepth: 2
+In this example, we will visualize a scaffold, modify it using user interface, and then export its nodal parameters.
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/visualising_scaffolds/visualising_scaffolds.rst]: modify it using user interface -> modify it using the existing user interface

File: docs/source/tutorials/visualising_scaffolds/visualising_scaffolds.rst [hunk 14]
```diff
@@ -1,21 +1,50 @@
...
 
-=======
-Case 1 :
-=======
+Once you have successfully installed and executed a new workflow, you should see ``scaffoldmaker`` window open.
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/visualising_scaffolds/visualising_scaffolds.rst]: you should see [the]

File: docs/source/tutorials/visualising_scaffolds/visualising_scaffolds.rst [hunk 15]
```diff
@@ -1,21 +1,50 @@
...
+In the left top corner, find :guilabel:`Mesh type` drop down list, and choose :guilabel:`3D Bladder with Urethra 1`.
+Often, there are more than one version of parameter set
```
Comment (PrasadBabarendaGamage) [inline docs/source/tutorials/visualising_scaffolds/visualising_scaffolds.rst]: of the parameter set

**Review by PrasadBabarendaGamage — CHANGES_REQUESTED** [review]
Excellent documentation @fpan017! Please see the minor comments attached.

**Pull request merged**